compiler/rustc_hir_analysis/src/hir_ty_lowering/bounds.rs

```diff
@@ -362,6 +362,7 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                         param_ty,
                         bounds,
                         predicate_filter,
+                        false,
```

Comment:
I don't like hard coding `false` here, we're in an extremely general method named `lower_bounds`. I'm not sure how likely it is that someone might call this with *not* wanting to accept overlapping assoc item constraints, but it atleast would be subtle enough I doubt people would realise they've done the wrong thing.

Can you try and explicitly thread the `overlapping_assoc_item_constraints` parameter up through the whole call site to all the places that actually know *why* the bounds are being lowered and so can reason about whether we should be accepting or rejecting overlapping assoc item constraints

Reply:
Maybe doing this will reveal that its way too much work to do so, or maybe it'll reveal how kinda pointless it is idk :3